crates/storage/engines/in_memory.rs

```diff
@@ -176,6 +176,28 @@ impl StoreEngine for Store {
         Ok(self.inner().transaction_pool.get(&hash).cloned())
     }
 
+    fn remove_transaction_from_pool(&self, hash: H256) -> Result<(), StoreError> {
+        self.inner().transaction_pool.remove(&hash);
+        Ok(())
+    }
+
+    fn filter_pool_transactions(
+        &self,
+        filter: &dyn Fn(&Transaction) -> bool,
+    ) -> Result<HashMap<Address, Vec<Transaction>>, StoreError> {
+        let mut txs: HashMap<Address, Vec<Transaction>> = HashMap::new();
+        for (_, tx) in self.inner().transaction_pool.iter() {
```

Comment:
Suggestion: `self.inner().transaction_pool.iter().filter(filter)`

Reply:
We are not only filtering here but also grouping transactions by sender, we could do it with filter_map and reduce or something similar but I think this way is more simple and readable

Reply:
I agree, the for is more simple and readable.
> We are not only filtering here but also grouping transactions by sender
Well, maybe we can rename txs to 'txs_by_sender'?

Reply:
sounds good